runtime/lib/app/activity-test-descriptor.js

```diff
@@ -0,0 +1,51 @@
+
+/**
+ * @namespace yodaRT.activity
+ */
+
+var inherits = require('util').inherits
+var EventEmitter = require('events').EventEmitter
+
+module.exports.ActivityTestDescriptor = ActivityTestDescriptor
+
+function ActivityTestDescriptor (activityDescriptor, appId, appHome, runtime) {
+  EventEmitter.call(this)
+  this._activityDescriptor = activityDescriptor
+  this._appId = appId
+  this._appHome = appHome
+  this._runtime = runtime
+}
+inherits(ActivityTestDescriptor, EventEmitter)
```

Comment:
Since most of constructor has to be rewritten and shares on base code, there shall be no base descriptor until class keyword has been introduced to ShadowNode.